frontend/BoardGUI.py

```diff
@@ -37,29 +41,101 @@ def build_board(self):
+            status = self.game_manager.remove_piece_mill(row, col)
+            if status:
+                print(f"Removed opponent's piece at ({row}, {col})")
+        else:
+            print(f"Invalid selection. Please select a piece from: {self.game_manager.removable_pieces}")
+    
+    def handle_moment_phase(self,row, col):
+        if self.game_manager.selected_piece is None:
+            # No piece selected, attempt to select a piece
+            piece_at_cell = self.game_manager.get_piece_at(row, col)
+            if piece_at_cell and piece_at_cell.get_state().name == self.game_manager.turn.name:
+                # Only allow selection if the piece belongs to the current player's turn
+                self.game_manager.selected_piece = (row, col)
+                print(f"Piece selected at ({row}, {col})")
+            else:
+                print("Cannot select piece: not current player's turn or empty cell")
+        else:
+            # A piece is already selected, attempt to move it
+            selected_row, selected_col = self.game_manager.selected_piece
+
+            # Check if the move is valid (adjacent and empty)
+            if self.game_manager.is_adjacent_and_empty(selected_row, selected_col, row, col):
+                # Move the piece to the new location
+                move_success = self.game_manager.move_piece(row, col)
+                if move_success:
+                    print(f"Piece moved to ({row}, {col})")
+                    # Remove the piece from the old position
+                    self.remove_piece(selected_row, selected_col)
+                    
+                    # Add the piece to the new position
+                    self.pieces_on_board[(row, col)] = self.game_manager.turn
+                    self.draw_board()  # Redraw board to show updated pieces
+                    
+                    # Check if the move forms a mill
+                    if self.game_manager.is_mill_formed(row, col):
+                        print("Mill formed! Remove opponent's piece.")
+                        self.game_manager.remove_opponent_piece(self.pieces_on_board)
+                    
+                    self.game_manager.end_turn()  # End turn after successful move
+                else:
+                    print(f"Invalid move to ({row}, {col})")
+            else:
+                print(f"Cell ({row}, {col}) is not adjacent or not empty")
+
+            # Clear the selected piece regardless of move success
+            self.game_manager.selected_piece = None
+    
+    
     def get_cell_clicked(self):
         current_cell = vec2(pg.mouse.get_pos()) // self.cell_size
         col, row = map(int, current_cell)
-        mouse_buttons = pg.mouse.get_pressed()  # Call this once and store the result
+        mouse_buttons = pg.mouse.get_pressed()
+        self.count += 1
+
+        # Handle opponent's piece removal if mill is formed
+        if self.game_manager.waiting_for_removal:
```

Comment:
Lets rename `self.game_manager.waiting_for_removal` to `self.game_manager.new_mill_is_formed` for better readability. Then we can remove the comment.

Reply:
hmm, variable should be named in such a way that indicates it is used in mill, so we can change